test/common/index.js

```diff
@@ -190,3 +195,60 @@ exports.runTestWithBuildType = async function (test, buildType) {
   await Promise.resolve(test(buildType))
     .finally(exports.mustCall());
 };
+
+// Some tests have to run in their own process, otherwise they would interfere
+// with each other. Such tests export a factory function rather than the test
+// itself so as to avoid automatic instantiation, and therefore interference,
+// in the main process. Two examples are addon and addon_data, both of which
+// use Napi::Env::SetInstanceData(). This helper function provides a common
+// approach for running such tests.
+exports.runTestInChildProcess = function ({ suite, testName, expectedStderr }) {
+  return exports.runTestWithBindingPath((bindingName) => {
+    return new Promise((resolve) => {
+      bindingName = escapeBackslashes(bindingName);
+      // Test suites are assumed to be located here.
+      const suitePath = escapeBackslashes(path.join(__dirname, '..', 'child_processes', suite));
+      const child = spawn(process.execPath, [
+        '--expose-gc',
+        '-e',
+        `require('${suitePath}').${testName}(require('${bindingName}'))`
+      ]);
+      let eventCount = 0;
+      const resultOfProcess = { stderr: [] };
+      const maybeResolve = (partialResult = {}) => {
+        Object.assign(resultOfProcess, partialResult);
+        if (++eventCount === 2) {
+          resolve(resultOfProcess);
+        }
+      };
+
+      // Capture the exit code and signal.
+      child
+        .on('exit', (code, signal) => maybeResolve({ code, signal }))
+        .on('close', () => maybeResolve());
```

Comment:
`'close'` event is always emitted after the `'exit'` event (https://nodejs.org/api/child_process.html#event-close). So this can be simplified as:

```suggestion
        .on('close', (code, signal) => resolve({ code, signal }));
```